src/Raven.Client/Documents/Session/DocumentSession.Attachments.cs

```diff
@@ -19,6 +21,35 @@ namespace Raven.Client.Documents.Session
     /// </summary>
     public partial class DocumentSession
     {
+        public AttachmentResult[] GetAttachmentNames<T>(string documentId)
+        {
+            var entity = Load<T>(documentId);
```

Comment:
Head doesn't return the metadata... so in case that this is the session already Load is a no-op.